packages/storage-driver-s3/src/index.ts

```diff
@@ -39,12 +39,16 @@ export class DriverS3 implements Driver {
 	private serverSideEncryption: string | undefined;
 
 	constructor(config: DriverS3Config) {
-		const s3ClientConfig: S3ClientConfig = {
-			credentials: {
+		const s3ClientConfig: S3ClientConfig = {};
+
+		if ((config.key && !config.secret) || (config.secret && !config.key)) {
+			throw new Error('Both `key` and `secret` are required when defined');
```

Comment:
Any reason we have to error out when only key or secret is configured through config? I could imagine a scenario where somebody might rely on AWS's default injection for SECRET, but keep the KEY in the config 🤔

Reply:
The TypeScript signature for `credentials` answers this:

```
Type '{ accessKeyId: string; }' is not assignable to type 'AwsCredentialIdentity | Provider<AwsCredentialIdentity>'.
  Property 'secretAccessKey' is missing in type '{ accessKeyId: string; }' but required in type 
```